src/MediatR/Mediator.cs

```diff
@@ -226,6 +256,25 @@ private IEnumerable<TWrapper> GetNotificationHandlers<TWrapper>(object notificat
                 .ToList();
         }
 
+        private Task<TResponse> GetPipeline<TResponse>(object request, RequestHandlerDelegate<TResponse> invokeHandler)
+        {
+            var requestType = request.GetType();
+            var method = CreatePipelineMethod.MakeGenericMethod(requestType, typeof(TResponse));

```

Comment:
I'm not sure any cache matters. With many containers (including the ASP.NET Core one), you have to register the Mediator class as scoped (see https://github.com/jbogard/MediatR.Extensions.Microsoft.DependencyInjection/blob/master/src/MediatR.Extensions.Microsoft.DependencyInjection/ServiceCollectionExtensions.cs#L188). So tbqh I'd be up for removing all the caches, it's just cognitive overhead.